src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -10,33 +10,24 @@ namespace Microsoft.DotNet.ApiCompatibility.Logging
     /// Suppression engine that contains a collection of <see cref="Suppression"/> items. It provides API to add a suppression, check if a passed-in suppression is already suppressed
     /// and serialize all suppressions down into a file.
     /// </summary>
-    public class SuppressionEngine : ISuppressionEngine
+    /// <param name="noWarn">A string that contains warning and error codes to ignore suppressions with the corresponding diagnostic id.</param>
+    /// <param name="baselineAllErrors">If true, baselines all errors.</param>
+    public class SuppressionEngine(string? noWarn = null, bool baselineAllErrors = false) : ISuppressionEngine
     {
         protected const string DiagnosticIdDocumentationComment = " https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids ";
-        private readonly HashSet<Suppression> _baselineSuppressions = new();
-        private readonly HashSet<Suppression> _suppressions = new();
-        private readonly HashSet<string> _noWarn;
+        private readonly HashSet<Suppression> _baselineSuppressions = [];
+        private readonly HashSet<Suppression> _suppressions = [];
+        private readonly HashSet<string> _noWarn = string.IsNullOrEmpty(noWarn) ? [] : new HashSet<string>(noWarn!.Split(';'));
```

Comment:
I'm surprised you had to use `noWarn!`. I would've assumed the nullability feature would know you already checked it.

Reply:
Yeah the `!` isn't necessary on .NETCoreApp but on .NETFramework which has inferior nullability compiler heuristics.